lib/can/include/CANMsg.h

```diff
@@ -1,13 +1,12 @@
 #pragma once
 
-#include <map>
-
+#include "LookupTable.h"
 #include "mbed.h"
 
 class CANMsg : public CANMessage {
  public:
   typedef mbed_error_status_t (*CANMsgHandler)(CANMsg &);
-  typedef std::map<unsigned int, CANMsg::CANMsgHandler> CANMsgHandlerMap;
+  typedef LookupTable<uint16_t, CANMsg::CANMsgHandler> CANMsgHandlerMap;
```

Comment:
rather than just a typedef, we make CANMSgHandler a subclass of LUT to always return a valid function with the default that just returns. this way the main apps can always just have `canHandlerMap[rxMsg.id](rxMsg);` with no additional error handling so that it appears a lot more straightforward at the high level app

Reply:
uhh not a fan of CANMsgHandler being a class. i think it makes more sense as an instance of a LUT. The caller could pass in an empty function as the default, which we discussed above. maybe it doesnt have to be a function, but rather a lambda expression? am i speaking garbage? ive never actually used em.

Reply:
lambdas are basically just anonymous functions. If you change the base LUT class to always return a default value, then this doesn't need to subclass. but what happens when theres no default provided and u make a call to []. does that mean the default value is mandatory?